lib/mayaUsd/listeners/notice.cpp

```diff
@@ -29,28 +29,32 @@ namespace {
 // to be removed.
 static int _newOrOpenRegistrationCount = 0;
 
-static void _OnMayaNewOrOpenSceneCallback(void* /*clientData*/)
+static void _afterNewSceneCallback(void* /*clientData*/) { UsdMayaSceneResetNotice().Send(); }
```

Comment:
Why do we need both a before new and after new callback?  What USD changes can occur between those two points in execution?

Reply:
Unfortunately, for the undo items clearing, we need to clear them earlier than when the after-new-scene gets triggered. But we also need to clear them for file-load and new-scene. You are right that for the undo items clearing, we would not need the after-new-scene callback, but they are all tied together now. As I said previously, the redundant 2nd call that will happen is harmless.